### ISSUE TITLE
Ownership error when uncompressing agent

### ISSUE THREAD

**Issue by epilgrim** [appsignal/appsignal-elixir#168]
Running inside docker containers (for development environment) the mix compile task fails after downloading the agent.
The problem is that tar tries to force the files to have the same user as they had before compressing them when it is run as root (which is the case inside docker by default).
```
/bin/tar: ./appsignal-agent: Cannot change ownership to uid 501, gid 50: Operation not permitted
/bin/tar: ./appsignal.h: Cannot change ownership to uid 501, gid 50: Operation not permitted
/bin/tar: ./appsignal.version: Cannot change ownership to uid 501, gid 50: Operation not permitted
/bin/tar: ./libappsignal.a: Cannot change ownership to uid 501, gid 50: Operation not permitted
/bin/tar: .: Cannot change ownership to uid 501, gid 50: Operation not permitted
/bin/tar: Exiting with failure status due to previous errors
==> appsignal
could not compile dependency :appsignal, "mix compile" failed. You can recompile this dependency with "mix deps.compile appsignal", update it with "mix deps.update appsignal" or clean it with "mix deps.clean appsignal"
==> sylar
** (Mix) Extracting of /tmp/appsignal-agent-557cdf6.tar.gz failed!
```

The tar utility has the `--no-same-owner` parameter to disable this behavior.
Could this parameter be added to the untar command? do we have other options?

**Comment by tombruijn** [comment]
Hi @epilgrim 

That's not good! It should use the `--no-same-owner` option by default according to `man tar`.

```
$ man tar
  ...

     --no-same-owner
           extract files as yourself (default for ordinary users)
```

What's your docker container setup like? What kind of user are you using in your docker container?
Let me know so I can reproduce the issue before patching it.

**Comment by epilgrim** [comment]
Hey @tombruijn 

Thanks for your promptly response.

The following is a GIST with the minimum files to reproduce the issue:
https://gist.github.com/epilgrim/55f68e44c8aea6269c73d621ecfddfd1

Important bits in my setup:
1. I'm using docker in MacOS, with the vm called Dinghy https://github.com/codekitchen/dinghy
2. The problem arises only if the file is being extracted into a volume (when extracting the `/tmp/appsignal-....` into /_build/....` which is shared to my host
3. What you say of the tar manpage is true, it is default, but for "ordinary" users. Docker containers in the simplest setup run as root, therefore the `--no-same-user` flag is not used

**Comment by epilgrim** [comment]
in any case, to circumvent the issue for now (and until I know what is your stance on this), what i'm doing is changing the user my container uses:
```
RUN useradd -ms /bin/zsh dev
USER dev
```
as the user is not root anymore, the default `--no-same-user` flag applies

**Comment by tombruijn** [comment]
Right.. Just tested this and can reproduce. But only with dinghy! The official docker for macOS app doesn't have this issue.

**Comment by tombruijn** [comment]
Hi @epilgrim I published a fix in PR #179. Can you try it out and let me know if you still run into the same issue?

**Comment by epilgrim** [comment]
hey! thanks.
yes, after that PR it works.

The problem with dinghy is because it adds a layer in the middle, mapping the user inside the vm to your local user, so you don't have all the files generated inside the VM as root in your host.

But the wrong owner should have happened anyways with docker for mac

in any case, thanks!!!
